indigo_api/views/documents.py

```diff
@@ -433,10 +435,23 @@ def post(self, request, document_id):
 
         # set this up to be the modified document
         remote_doc = Document.objects.get(pk=local_doc.pk)
-        serializer.fields['document'].update_document(local_doc, serializer.validated_data['document'])
 
+        # this will set the local_doc's content as the <portion> in provision mode,
+        # and update it with the latest unsaved changes regardless
+        serializer.set_content()
         local_doc.content = differ.preprocess_xml_str(local_doc.document_xml).decode('utf-8')
-        remote_doc.content = differ.preprocess_xml_str(remote_doc.document_xml).decode('utf-8')
+
+        provision_eid = serializer.validated_data.get('provision_eid')
+        if provision_eid:
+            portion = remote_doc.get_portion(provision_eid)
+            # the same structure as the 'xml' we're getting from the browser: akn/portion/portionBody/element
+            remote_xml = etree.tostring(portion.root, encoding='unicode')
+            remote_doc.work.work_uri.doctype = 'portion'
+            remote_doc.content = differ.preprocess_xml_str(remote_xml)
+
+        else:
+            # full document mode
```

Comment:
Why does this call decode but the call a few lines up doesn't?

Reply:
It looks like at the end of the day that cobalt's `AkomaNtosoDocument` class will encode a `str` back into `bytes`, and `preprocess_xml_str` returns `bytes`. So actually you don't need any of the `decode()`s when setting `.content = `.